Sources/PrimerSDK/Classes/Extensions & Utilities/StringExtension.swift

```diff
@@ -169,16 +169,23 @@ internal extension String {
         return isValid
     }
     
-    var isTypingValidCardholderName: Bool? {
-        if isValidCardholderName { return true }
-        return nil
+    var isTypingNonDecimalCharacters: Bool {
+        isValidNonDecimalString
     }
     
-    var isValidCardholderName: Bool {
+    var isValidNonDecimalString: Bool {
         if isEmpty { return false }
         return rangeOfCharacter(from: .decimalDigits) == nil
     }
     
+    var isValidCardholderName: Bool {
+        return isValidNonDecimalString
```

Comment:
Can you review those names and make them more declarative? Or if they are declarative, can you review the `isValidCardholderName` functionality?

I would expect `isValidNonDecimalString` to allow `44o!` to pass, but this is not a valid cardholder name.

Reply:
I reviewed the logic from within the:

```
func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
```

of each FieldView where it is appropriate to make a better check as we type.
The `PrimerSimpleCardFormTextFieldView` will still allow decimal inputs.